src/wasm.rs

```diff
@@ -112,6 +125,15 @@ impl CompiledWasmModule {
       std::slice::from_raw_parts(ptr, len.try_into().unwrap())
     }
   }
+
+  pub fn source_url(&self) -> &str {
```

Comment:
This returns a &str on the C++ heap with infinite lifetime. This is definitely unsafe - reverting this PR.